python-package/xgboost/core.py

```diff
@@ -637,15 +631,14 @@ def set_uint_info(self, field, data):
         data: numpy array
             The array of data to be set
         """
-        data = np.array(data, copy=False, dtype=ctypes.c_uint)
-        c_data = c_array(ctypes.c_uint, data)
-        length, stride = _get_length_and_stride(data)
-        _check_call(_LIB.XGDMatrixSetUIntInfoStrided(
-            self.handle,
-            c_str(field),
-            c_data,
-            c_bst_ulong(stride),
-            c_bst_ulong(length)))
+        if getattr(data, 'base', None) is not None and \
+           data.base is not None and isinstance(data, np.ndarray) \
+           and isinstance(data.base, np.ndarray) and (not data.flags.c_contiguous):
+            raise ValueError('Sliced data is not allowed for uint property')
```

Comment:
Why not?

Reply:
Right now, there is only one `uint` type field called `root_index`. Do we need slicing for this?

Reply:
We are here anyway, no need to leave something can be done in one line I think.  But go as you think best.

Reply:
Addressed in the latest commit